client/apps/game/src/ui/modules/ws-chat/chat.tsx

```
@@ -629,31 +553,133 @@ function ChatModule() {
     return filteredUsers
       .filter((user) => user && user?.id !== userId && !(unreadMessages[user?.id] || 0))
-      .sort((a, b) => (a?.username || a?.id).localeCompare(b?.username || b?.id));
-  }, [filteredUsers, userId, unreadMessages]);
+      .sort((a, b) => {
+        const isPinnedA = pinnedUsers.includes(a?.id || "");
+        const isPinnedB = pinnedUsers.includes(b?.id || "");
+
+        if (isPinnedA && !isPinnedB) return -1;
+        if (!isPinnedA && isPinnedB) return 1;
+
+        return (a?.username || a?.id).localeCompare(b?.username || b?.id);
```

Biome (1.9.4)

[error] Unsafe usage of optional chaining.

If it short-circuits with 'undefined' the evaluation will throw TypeError here:


(lint/correctness/noUnsafeOptionalChaining)